shared/team-building/user-result.native.tsx

```diff
@@ -0,0 +1,210 @@
+import LegacyUserResult from './user-result.desktop'
+
+// TODO
+// * Use ListItem2
+
+class UserResult extends React.Component<Props> {
+  render = () => {
+    const keybaseResult = this.props.resultForService === 'keybase'
+    const keybaseUsername: string | null = this.props.services['keybase'] || null
+    const serviceUsername = this.props.services[this.props.resultForService]
+    const onAdd = !this.props.isPreExistingTeamMember ? this.props.onAdd : undefined
+    const onRemove = !this.props.isPreExistingTeamMember ? this.props.onRemove : undefined
+
+    return (
+      <Kb.ClickableBox onClick={this.props.inTeam ? onRemove : onAdd}>
+        <Kb.Box2
+          className={Styles.classNames({
+            hoverRow: !this.props.inTeam,
+            hoverRowinTeam: this.props.inTeam,
+          })}
+          direction="horizontal"
+          fullWidth={true}
+          centerChildren={true}
+          style={styles.rowContainer}
+        >
+          <Avatar resultForService={this.props.resultForService} keybaseUsername={keybaseUsername} />
+          <Username
+            isPreExistingTeamMember={this.props.isPreExistingTeamMember}
+            keybaseResult={keybaseResult}
+            keybaseUsername={keybaseUsername}
+            displayLabel={this.props.displayLabel}
+            username={serviceUsername || ''}
+            prettyName={this.props.prettyName}
+            followingState={this.props.followingState}
+            services={this.props.services}
+          />
+          {!this.props.isPreExistingTeamMember && (
+            <ActionButton
+              inTeam={this.props.inTeam}
+              onAdd={this.props.onAdd}
+              onRemove={this.props.onRemove}
+              highlight={this.props.highlight}
+            />
+          )}
+        </Kb.Box2>
+      </Kb.ClickableBox>
+    )
+  }
+}
+
+const AvatarSize = 48
+const Avatar = ({
+  resultForService,
+  keybaseUsername,
+}: {
+  keybaseUsername: string | null
+  resultForService: Types.ServiceIdWithContact
+}) => {
+  if (keybaseUsername) {
+    return <Kb.Avatar size={AvatarSize} username={keybaseUsername} />
+  }
+
+  return (
+    <Kb.Icon
+      fontSize={AvatarSize}
+      type={serviceIdToIconFont(resultForService)}
+      colorOverride={serviceIdToAccentColor(resultForService)}
+    />
+  )
+}
+
+const isPreExistingTeamMemberText = (prettyName: string) =>
+  `${prettyName ? prettyName + ' • ' : ''} Already in team`
+
+const FormatPrettyName = (props: {
+  followingState: Types.FollowingState
+  keybaseResult: boolean
+  keybaseUsername: string | null
+  prettyName: string
+  services: [Types.ServiceIdWithContact]
+}) =>
+  props.keybaseResult ? (
+    <Kb.Box2 direction="horizontal" fullWidth={true} style={styles.services}>
+      {!!props.prettyName && (
+        <Kb.Text type="BodySmall">{props.prettyName + (props.services.length ? ' •' : '')}</Kb.Text>
+      )}
+      {props.services.map(service => (
```

Comment:
instead of the `as` below, can you `map<Types.ServiceIdWithContact>`?

Reply:
i'm confused why you need either. TS should know the type